presto-password-authenticators/src/test/java/io/prestosql/plugin/password/salesforce/TestSalesforceBasicAuthenticator.java

```diff
@@ -0,0 +1,269 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.plugin.password.salesforce;
+
+import com.google.common.net.MediaType;
+import io.airlift.http.client.HttpClient;
+import io.airlift.http.client.HttpStatus;
+import io.airlift.http.client.jetty.JettyHttpClient;
+import io.airlift.http.client.testing.TestingHttpClient;
+import io.prestosql.spi.security.AccessDeniedException;
+import org.testng.SkipException;
+import org.testng.annotations.BeforeSuite;
+import org.testng.annotations.Test;
+
+import java.security.Principal;
+
+import static io.airlift.http.client.testing.TestingResponse.mockResponse;
+import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.fail;
+
+public class TestSalesforceBasicAuthenticator
+{
+    private HttpClient testHttpClient;
+    private String org;
+    private String username;
+    private String password;
+    private String successResponse;
+    private String failedResponse;
+    private SalesforceConfig config;
```

Comment:
Presto runs test in parallel with multiple threads, and I don't think this is thread safe.

I suggest removing all of the fields here and up, and just using local variables.